### PR TITLE
feat(server): process post request body

### PR DESCRIPTION
**Pull request by rofe**
#179

### CODE REVIEW

**Comment by rofe** [comment]
I will add tests once we have agreement on approach.

**Comment by tripodsan** [comment]
Why use express.json when stringify it again? Use express.bodsyparser-raw instead.

**Comment by tripodsan** [comment]
Also. I order to emulate openwhisk correctly, use `_ow_body` instead

File: src/HelixServer.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ function executeTemplate(ctx) {
     selector: ctx._selector,
     extension: ctx._extension,
     rootPath: ctx._mount,
+    body: ctx._body,
```
Comment (tripodsan) [inline src/HelixServer.js]: it should populate all properties in `ctx._body`, eg:

```
const params = Object.assign({ .. }, ctx._body);
```

File: src/RequestContext.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ module.exports = class RequestContext {
     this._extension = '';
     this._headers = req.headers || {};
     this._method = req.method || 'GET';
+    this._body = JSON.stringify(req.body);
```
Comment (tripodsan) [inline src/RequestContext.js]: keep it as json, like openwhisk does.

Comment (rofe) [reply src/RequestContext.js]: 👍

File: src/HelixServer.js (hunk 1 above)
Comment (rofe) [reply src/HelixServer.js]: instead of passing the body, I will add all top level items in the request body to the context.

**Comment by rofe** [comment]
At the moment, I'm trying to reproduce this in runtime without any changes in the pipeline...

**Comment by rofe** [comment]
Together with https://github.com/adobe/helix-pipeline/pull/243, this will allow posting `content.body`, which will in turn skip the fetch step in the pipeline.

**Pull request merged**